front-end/src/app/shared/components/inputs/loan-terms-dates-input/loan-terms-dates-input.component.ts

```diff
@@ -142,24 +142,30 @@ export class LoanTermsDatesInputComponent extends BaseInputComponent implements
             initialSelectionStart - lengthDifference,
             initialSelectionEnd - lengthDifference,
           );
+          interestField.markAsTouched();
         }
 
         if (newInterestRate.length > 0) {
           if (!newInterestRate.endsWith('%')) {
             interestField.setValue(newInterestRate + '%');
-
             textInput?.setSelectionRange(newInterestRate.length, newInterestRate.length);
+            interestField.markAsTouched();
           }
         }
         if (interestField.value === '%') {
           interestField.setValue('');
         }
       }
-
-      interestField.markAsTouched();
     }
   }
 
+  /**
+   * Alternates between a date form control and a string form control.
+   * The new CalendarComponent replaces the string form control with a Date control.
+   * This happens on Init, which means when the ngIf condition is met.
+   * This means when returning to the string control we need to recreate the control.
+   * @param newDueDateSetting
+   */
```

Comment:
This conversion should no longer happen.  Switching the type should clear the value https://fecgov.atlassian.net/browse/FECFILE-1786

Reply:
Ah, I totally misread that. I thought it was specifically talking about only the error messages being cleared, not the values themselves as well. I made some updates so now both the due date field and the interest rate field get set to empty when the associated setting field is changed.